tests/integration/dynamic-machines-test.ts

```diff
@@ -0,0 +1,77 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { click, render } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+
+import { assign, createMachine, spawn } from 'xstate';
+
+import type { Interpreter } from 'xstate';
+
+module('Dynamic Machines', function (hooks) {
+  setupRenderingTest(hooks);
+
+  test('Can have a dynamically created machine and is reactive', async function (assert) {
+    function createNested(context: Record<string, unknown>) {
+      return createMachine({
+        id: 'nested',
+        initial: 'inactive',
+
+        context,
+
+        states: {
+          inactive: { on: { TOGGLE: 'active' } },
+          active: { on: { TOGGLE: 'inactive' } },
+        },
```

Comment:
Can you please include actions here, that would be run outside of the machine? Main trick that I've wanted to achieve, to figure out how we can have actions in nested machines that are handled by component.

Reply:
Yeah, I can give it a go! ✨